### PR TITLE
[stable10] Don't check for Same-Site cookie on Chrome Android

### PR DESCRIPTION
**Pull request by LukasReschke**
Backport of https://github.com/nextcloud/server/pull/1454

### CODE REVIEW

**Comment by MorrisJobke** [comment]
This breaks login and the public share page.

**Comment by MorrisJobke** [comment]
👎

**Comment by LukasReschke** [comment]
> This breaks login and the public share page.

This is the same code than on master, so I suspect this also to be not properly working in Android then. My guess is that the cookies are simply never sent then.

@MorrisJobke Can you add a `self::sendSameSiteCookies();` before the return in line 529? I suppose that way it should work then… *Should*, as I don't have an Android device to test this myself 🙈

**Comment by MorrisJobke** [comment]
> This is the same code than on master, so I suspect this also to be not properly working in Android then. My guess is that the cookies are simply never sent then.

But I applied the patch also on my productive instance ... weird

**Comment by MorrisJobke** [comment]
> This breaks login and the public share page.

I retested this and it works fine now. I guess this was a non-stable10 compatible app back then.

:+1: from me

**Comment by MorrisJobke** [comment]
@nickvergessen Please test this :)

* create public link share of a folder with an image in it
* open this link in Chrome on Android
* download of image should succeed

File: lib/base.php [hunk 1]
```diff
@@ -520,6 +520,14 @@ private static function performSameSiteCookieProtection() {
 			return;
 		}
 
+		// Chrome on Android has a bug that it doesn't sent cookies with the
+		// same-site attribute for the download manager. To work around that
+		// all same-site cookies get deleted and recreated directly. Awesome!
+		// FIXME: Remove once Chrome 54 is deployed to end-users
+		// @see https://github.com/nextcloud/server/pull/1454
+		if($request->isUserAgent([\OC\AppFramework\Http\Request::USER_AGENT_ANDROID_MOBILE_CHROME])) {
+			return;
```
Comment (LukasReschke) [inline lib/base.php]: I guess I also should move the logic from "self::sendSameSiteCookies();" above this so it catches some more cases. Will correct later.

**Comment by nickvergessen** [comment]
status @LukasReschke ?

**Comment by LukasReschke** [comment]
Chrome 54 is out for mobile => Unnecessary.

**Pull request closed without merging**